src/components/DetailsList/DetailsList.tsx

```diff
@@ -224,7 +225,7 @@ export class DetailsList extends React.Component<IDetailsListProps, IDetailsList
         data-is-scrollable='false'
         aria-label={ ariaLabel }
         role={ shouldApplyApplicationRole ? 'application' : '' }>
-        <div role='grid' aria-label={ ariaLabelForGrid }>
+        <div role={ shouldApplyGridRole ? 'grid' : '' } aria-label={ ariaLabelForGrid }>
```

Comment:
Please just either set role correctly, or let the caller set role.

We need to avoid all flags that are need in depth explanation. As a user I have no idea what these mean, or why I should care.

What is the correct value here?